runtime/libpgmath/lib/common/dispatch.c

```diff
@@ -1264,7 +1281,8 @@ __math_dispatch_init()
   if (__sync_bool_compare_and_swap(&__math_dispatch_in_prog, false, true)) {
```

Comment:
Just a quick comment not about this commit but in general -- shouldn't this be a loop?

Reply:
In theory, the builtin __sync_bool_compare_and_swap should loop, if and only if it's implemented as LL/SC and not as a lock-free atomic.

 In AArch64 for T99, we support LSE - which means it's a lock-free atomic. I'm pretty sure it's the same on Intel X86_64 and PPCle. 

If the builtin is written as LL/SC and doens't loop, that's a builtin bug.